packages/embark/src/lib/modules/ipfs/index.js

```diff
@@ -25,44 +25,64 @@ class IPFS {
     this.storageProcessesLauncher = null;
     this.usingRunningNode = false;
     this.modulesPath = dappPath(embark.config.embarkConfig.generationDir, constants.dappArtifacts.symlinkDir);
+    this.registered = false;
 
     this.webServerConfig = embark.config.webServerConfig;
     this.blockchainConfig = embark.config.blockchainConfig;
 
+    this.embark.events.setCommandHandler("module:ipfs:reset", (cb) => {
+      this.events.request("processes:stop", "ipfs", (err) => {
+        if (err) {
+          this.logger.error(__('Error stopping IPFS process'), err);
+        }
+        this.init(cb);
+      });
+    });
+
+    this.init();
+  }
+
+  init(callback = () => {}) {
     if (!this.isIpfsStorageEnabledInTheConfig()) {
-      return this.events.emit("ipfs:process:started", null, false);
+      this.events.emit("ipfs:process:started", null, false);
+      return callback();
     }
-
-    this.setServiceCheck();
-    this.registerUploadCommand();
-    this.listenToCommands();
-    this.registerConsoleCommands();
-    this.events.request("processes:register", "ipfs", {
-      launchFn: (cb) => {
-        if(this.usingRunningNode) {
-          return cb(__("IPFS process is running in a separate process and cannot be started by Embark."));
-        }
-        this.startProcess((err, newProcessStarted) => {
-          this.addStorageProviderToEmbarkJS();
-          this.addObjectToConsole();
-          this.events.emit("ipfs:process:started", err, newProcessStarted);
-          cb();
-        });
-      },
-      stopFn: (cb) => {
-        if(this.usingRunningNode) {
-          return cb(__("IPFS process is running in a separate process and cannot be stopped by Embark."));
+    if (!this.registered) {
+      this.registered = true;
```

Comment:
Probably not a biggie, but maybe we want to set `registered = true` when we indeed know that it has been registered (inside the `launchFn()`)

Reply:
Actually, putting it inside the `launchFn` would be bad, because it's not a callback for the registration but a call back for when the launch process is called and in my case, `registered` is really for when the events are registred. Open to siggestions if you think the name is confusing

Reply:
Okay, fair enough! :)